crates/wassette/src/lib.rs

```diff
@@ -198,577 +300,241 @@ pub struct ComponentInstance {
 }
 
 impl LifecycleManager {
-    /// Creates a lifecycle manager from configuration parameters
-    /// This is the primary way to create a LifecycleManager for most use cases
+    /// Begin constructing a lifecycle manager with a fluent builder that
+    /// validates configuration and applies sensible defaults.
+    pub fn builder(plugin_dir: impl Into<PathBuf>) -> LifecycleBuilder {
```

Comment:
Nit: `Into<PathBuf>` looks a bit odd to me especially because all your calls are doing `to_path_buf` below. This should probably just be `AsRef<Path>` and then do the conversion internally.